packages/dbos-cloud/users/login.ts

```diff
@@ -45,3 +61,202 @@ export async function login(host: string, getRefreshToken: boolean, useRefreshTo
   }
   return 0;
 }
+
+/**
+ * Login and obtain user credentials.
+ * First checks if the credentials exist and not expired.
+ * If so, returns the credentials.
+ * If not, try to login the user.
+ * If the user is not registered, prompts to register the user.
+ * @param {string} host - The DBOS Cloud host to authenticate against.
+ * @returns {DBOSCloudCredentials} - The user's DBOS Cloud credentials.
+ */
+export async function loginGetCloudCredentials(host: string, logger: Logger): Promise<DBOSCloudCredentials> {
+  // Check if credentials exist and are not expired
+  const credentials = await checkCredentials(logger);
+
+  if (credentials.token !== "" && credentials.userName !== "") {
+    logger.debug(`Logged in as ${credentials.userName}`);
+    writeCredentials(credentials);
```

Comment:
Why do we need to write credentials here?

Reply:
Done.